DSharpPlus/Clients/DiscordClient.cs

```diff
@@ -1215,6 +1215,37 @@ private void PopulateMessageReactionsAndCache(DiscordMessage message, TransportU
         }
     }
 
+    // Ensures the channel is cached:
+    // - DM -> _privateChannels dict on DiscordClient
+    // - Thread -> DiscordGuild#_threads
+    // - _ -> DiscordGuild#_channels
+    private void UpdateChannelCache(DiscordChannel? channel)
+    {
+        if (channel is null)
+        {
+            return;
+        }
+        
+        if (channel.GuildId is null)
+        {
+            if (channel is DiscordDmChannel dmChannel)

```

Comment:
Are both of these checks really necessary? You could just switch on the channel type

Reply:
Theoretical should one of those be enough, yeah

Reply:
My recommendation would be just to check for the type, since `DiscordDmChannel` implicates the guild id being null anyway

Reply:
yeah currently working on it